packages/devtools_app/lib/src/screens/app_size/app_size_controller.dart

```diff
@@ -279,8 +279,28 @@ class AppSizeController {
 
     Map<String, dynamic> diffMap;
     if (oldFile.isAnalyzeSizeFile && newFile.isAnalyzeSizeFile) {
+      var oldFileJson = oldFile.data as Map<String, dynamic>;
+      var newFileJson = newFile.data as Map<String, dynamic>;
+
+      if (!(oldFileJson['n'] == 'ArtificialRoot') ==
```

Comment:
Let's pull the check for the `ArtificialRoot` into a helper function called `isDeferredFormat` which takes in the processed JSON and returns a bool. We can then use that here, and in the `isDeferredApp` check above.

Reply:
Let's also assign it to a constant variable (eg `artificialRootName`, like this: https://github.com/flutter/devtools/blob/master/packages/devtools_app/lib/src/screens/debugger/debugger_screen.dart#L102) so that we don't have to use the raw string everywhere.